platform/locale.mm

```diff
@@ -9,6 +9,8 @@ Locale GetCurrentLocale()
   NSLocale * locale = [NSLocale currentLocale];
   return {locale.languageCode ? [locale.languageCode UTF8String] : "",
           locale.countryCode ? [locale.countryCode UTF8String] : "",
-          locale.currencyCode ? [locale.currencyCode UTF8String] : ""};
+          locale.currencyCode ? [locale.currencyCode UTF8String] : "",
+          locale.decimalSeparator ? [locale.decimalSeparator UTF8String] : ".",
+          locale.groupingSeparator ? [locale.groupingSeparator UTF8String] : ","};
```

Comment:
I'm not sure that this default value is OK. I think, it's better to leave it empty.

Reply:
Yes, maybe an empty grouping separator may be more common across different locales, though this default value will be overwritten by system's configuration in most of the times.

Your comment also confirms that in many languages, the grouping separator is only used in values with more than 4 digits, keeping digits with 4 digits with no grouping separator ("10 123 m" and "1234 m"). Maybe we shall also keep this behavior, which is also _suggested_ as International Standard according to [this](https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping) Wikipedia article about digit grouping. This International Standard also _suggests_ to use always the narrow no-break space char as the grouping separator.

My proposal for the grouping separator would be then: set empty "" (no grouping separator) as default if the system does not provide any & apply grouping separators *only* on numbers with more that 4 digits.

Reply:
Thanks! Looks like a good default is:
- Use group separator for >4 chars only
- Use a non-breaking space

P.S. Is non-breaking space used after a number and before the unit name? Should we use it too?

Reply:
@biodranik: I assume that you suggest to use as default the **narrow** non-breaking space (NNBSP, U+202F), instead of the **standard** non-breaking space (NBSP, U+00A0).

In [this](https://en.wikipedia.org/wiki/Non-breaking_space#Uses_and_variations) Wikipedia article, it's indeed stated that the NBSP's are also used between the value and the units, so yes, we can also use them there instead of the regular whitespace.

In a first commit I've set the NNBSP as default for iOS and linux and I've modified the UTs accordingly, but after pushing the commit, the iOS tests have failed because the grouping separator is configured to the ',' character. The workaround is to set the default character in linux and the UTs with the ',' character.

I've amended the commit according to these last comments.